lsp-go.el

```diff
@@ -120,6 +119,33 @@ defaults to half of your CPU cores."
   :risky t
   :type '(repeat string))
 
+(defcustom lsp-clients-go-library-directories-include-go-modules t
+  "Whether or not $GOPATH/pkg/mod should be included as a library directory."
+  :type 'boolean
+  :group 'lsp-clients-go)
+
+(defun lsp-clients-go--library-default-directories (_workspace)
+  "Calculate go library directories.
+
+If `lsp-clients-go-library-directories-include-go-modules' is non-nil
+and the environment variable GOPATH is set this function will return
+$GOPATH/pkg/mod along with the value of
+`lsp-clients-go-library-directories'."
+  (let ((library-dirs lsp-clients-go-library-directories))
+    (when (and lsp-clients-go-library-directories-include-go-modules
+               (or (and (not (file-remote-p default-directory)) (executable-find "go"))
+                   (and (version<= "27.0" emacs-version) (with-no-warnings (executable-find "go" (file-remote-p default-directory))))))
+              (with-temp-buffer
+                (when (= 0 (call-process "go" nil t nil "env" "GOPATH"))
```

Comment:
nit: use zerop, also you may consider using `shell-command-to-string`

Reply:
Does shell-command-to-string work properly for remote buffers?

I'll switch this to use zerop.

Reply:
Not sure, if it doesn't I am fine with the code as it is.

Reply:
I'll do some testing with tramp and let you know.

Reply:
Just to clarify - I am fine to have it as it is even without testing whether it works over tramp. It is up to you whether you want to invest more time into that...

Reply:
I've updated the pr with proper support for remote sessions. (Remote sessions is one of the biggest complaints in go-mode.el with the legacy non-lsp tools, so we really want it to work for people using gopls).